src/eradiate/contexts.py

```diff
@@ -224,6 +224,19 @@ class CKDSpectralContext(SpectralContext):
         doc="The bindex value corresponding to this spectral context. "
         "The default value is a simple placeholder used for testing purposes.",
         type=":class:`.Bindex`",
+        init_type=":class:`.Bindex` or tuple or dict, optional",
+    )
+
+    bin_set: t.Optional[BinSet] = documented(
+        attr.ib(
+            default=None,
+            converter=attr.converters.optional(BinSet.convert),
+            validator=attr.validators.optional(attr.validators.instance_of(BinSet)),
+        ),
+        doc="If relevant, the bin set from which ``bindex`` originates.",
```

Comment:
Why wouldn't it be relevant? I thought a bindex was ambiguous without the reference to the bin set.

Reply:
Unless you have to select a bin set-dependent data file, you don't care about the bin set ID. Most components can actually evaluate optical properties in CKD modes without information about the bin set ID, gaseous absorption data is a notable exception. So I see no reason to always require bin set information when creating a `CKDSpectralContext` object.

Reply:
Ah that's right, I forgot about that